docs/extend/extend-cli/cli-developing-a-plugin.md

```diff
@@ -2,218 +2,339 @@
+2. In the `diff` directory, create a file named `Diff.definition.ts`.
+3. Copy and paste the following code into the `Diff.definition.ts` file:
+
+    ```typescript
+    import { ICommandDefinition } from "@zowe/imperative";
+    import { DataSetsDefinition } from "./data-sets/DataSets.definition";
+    const IssueDefinition: ICommandDefinition = {
+        name: "diff",
+        summary: "Diff two data sets content",
+        description: "Uses open source diff packages to diff two data sets content",
+        type: "group",
+        children: [DataSetsDefinition]
+    };
+
+    export = IssueDefinition;
+    ```
+
+4. In the `diff` folder, create a folder named `data-sets`. 
+5. In the `data-sets` folder, create the following two files: 
+    - `DataSets.definition.ts`
+    - `DataSets.handler.ts`
+
+6. Copy and paste the following code into the `DataSets.definition.ts` file:
+    ```typescript
+    import { ICommandDefinition } from "@zowe/imperative";
+
+    export const DataSetsDefinition: ICommandDefinition = {
+        name: "data-sets",
+        aliases: ["ds"],
+        summary: "data sets to diff",
+        description: "diff the first data set with the second",
+        type: "command",
+        handler: __dirname + "/DataSets.handler",
+        positionals: [
+            {
+                name: "oldDataSet",
+                description: "The old data set",
+                type: "string"
+            },
+            {
+                name: "newDataSet",
+                description: "The new data set",
+                type: "string"
+            }
+        ],
+        profile: {
+            required: ["zosmf"]
+        }
+    };
+    ```
+
+7. Copy and paste the following code into the `DataSets.handler.ts` file:
+    ```typescript
+    import { ConnectionPropsForSessCfg, ICommandHandler, IHandlerParameters, ISession, Session } from "@zowe/imperative";
+    import { DataSetDiff } from "../../../api/DataSetDiff";
+
+    export default class DataSetsDiffHandler implements ICommandHandler {
+        public async process(params: IHandlerParameters): Promise<void> {
+            const sessCfg: ISession = {
+                hostname: params.arguments.host,
+                port: params.arguments.port,
+                basePath: params.arguments.basePath,
+                rejectUnauthorized: params.arguments.rejectUnauthorized,
+                protocol: params.arguments.protocol || "https"
+            };
+            const sessCfgWithCreds = await ConnectionPropsForSessCfg.   addPropsOrPrompt<ISession>(sessCfg, params.arguments,
+                { doPrompting: true, parms: params });
+            const session = new Session(sessCfgWithCreds);
+
+            const resp = await DataSetDiff.diff(session, params.arguments.oldDataSet, params.arguments.newDataSet);
+            params.response.console.log(resp);
+        }
+    }    
+    ```
+    When successful, **[WHAT IS THE USER SUPPOSED TO SEE?]**.
```

Comment:
similar to previous comments...
instruction on how to test OR remove this `when successful`